modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCGroupByVisitorOnlineTest.java

```diff
@@ -109,11 +111,53 @@ public void testComputeOnMathExpressionWithLimit() throws Exception {
         PropertyName pn = ff.property(aname("energy_consumption"));
         Multiply computeAttribute = ff.multiply(pn, ff.literal(10));
         PropertyName groupAttribute = ff.property(aname("building_type"));
-        List<Object[]> value  = genericGroupByTestTest(queryWithLimits(3, 2), Aggregate.MAX, computeAttribute, true, groupAttribute);
+        List<Object[]> value = genericGroupByTestTest(queryWithLimits(3, 2), Aggregate.MAX, computeAttribute, true, groupAttribute);
         assertNotNull(value);
 
         assertTrue(value.size() >= 1 && value.size() <= 2);
     }
+    
+    public void testNumericHistogram() throws Exception {
+        // buckets with a size of 100, the function returns an integer from 0 onwards, which
+        // is a zero based bucket number in the bucket sequence 
+        FilterFactory ff = dataStore.getFilterFactory();
+        PropertyName pn = ff.property(aname("energy_consumption"));
+        Expression expression = ff.function("floor", ff.divide(pn, ff.literal(100)));
+
+        boolean expectOptimized = dataStore.getFilterCapabilities().supports(FilterFunction_floor.class);
+        List<Object[]> value  = genericGroupByTestTest(Query.ALL, Aggregate.COUNT, expectOptimized, expression);
+        assertNotNull(value);
+
+        assertEquals(value.size(), 3);
+        checkValueContains(value, "0", "10");
+        checkValueContains(value, "1", "1");
+        checkValueContains(value, "5", "1");
```

Comment:
Bucket for ``500`` value is ``5`` since bucket ``4`` range goes from ``400-499`` right ? This clear in documentation , but I would complete this test comment with something like:

```
// buckets with a size of 100, the function returns an integer from 0 onwards (0-99), which
// is a zero based bucket number in the bucket sequence 
```